app/tests/unit/pages/project-revision/[projectRevision]/index.test.tsx

```diff
@@ -238,4 +239,24 @@ describe("The Create Project page", () => {
       true
     );
   });
+
+  it("redirects to the 404 page when a revision doesn't exist", async () => {
+    const spy = jest
+      .spyOn(require("app/hooks/useRedirectTo404IfFalsy"), "default")
+      .mockImplementation(() => {
+        return true;
+      });
```

Comment:
why does this need to be mocked? Is the revision not actually `undefined`? If that is the case that would explain why you were not catching issues with using the undefined variable, and using a mock here is harmful to your test, as you're making your function take a path that it wouldn't otherwise.

Reply:
From what I saw the relay testing framework puts mock values wherever it's not defined in the generator configuration that we pass. 
I think if we want relay to return `undefined` values, we'll need to pass `null` somewhere  in the call to `MockPayloadGenerator.generate(...)`, and grab the result in another `const nullInitialQueryRef = loadQuery<ProjectRevisionQuery>(...)` object that we can use.

@BCerki happy to pair on that if you want, or if what I'm saying makes no sense

Reply:
Here I'm only trying to test that the component renders null (sorry for the misleading test name, forgot to change it) if the revision doesn't exist, same as I'm doing in `app/tests/unit/pages/operator/[operator].test.tsx`, etc. I'm trying to test the redirect part in the hook test.

Reply:
Sure, that would be great, thanks